inductiva/resources/machine_groups.py

```diff
@@ -133,3 +132,34 @@ def get():
         machine_group_list.append(mg_class.from_api_response(mg))
 
     return machine_group_list
+
+
+def get_machine_type(num_cpus: int,
+                     ram_gb: Optional[int] = None,
+                     spot: bool = False):
```

Comment:
Does spot matter for the calculation? For example, if we have two machines, A and B, where A is cheaper than B without spot, can it occur that B will be cheaper than A if both have spot?

Or do you just want to give the cost at this moment?

Reply:
If I understand your question correctly, in theory a machine with the roughly same config cannot be cheaper than a similar machine with spot, because spot should be A LOT cheaper.  Here I wanted to explicitly separate spot and non-spot machine types, so that the users are mindful about the machine price being calculated in spot. Later if you want to provide both options, you can alway call this method twice and give the user spot and non-spot prices, that's a matter of deciding what the user interface should look like.

Reply:
My question is that between two machines, the same will always be cheapest, no matter if both are spot or not (which is your comparison both are or both aren't), because the discount is proportional to each or not?

My concern, is that they set this here to get the machine type, and then they will need to add spot again in the MachineGroup creation.

Reply:
I just tested it and with the same input the function returned two different machine types depending on spot True or False, so the machine types will be different. 

The second part is yes, they will need to configure MachineGroup accordingly.

Reply:
Interesting! Thanks for testing that!